File: tasks/2025/round4_pp/twolanes-pp.py
#!/usr/bin/env python3
# @check-accepted: examples N<=15 lane1=lane2 oneObstacle no-limits
# NOTE: it is recommended to use this even if you don't understand the following code.

import sys

# uncomment the two following lines if you want to read/write from files
# sys.stdin = open('input.txt')
# sys.stdout = open('output.txt', 'w')

N = int(input().strip())

lane1 = input().strip()
lane2 = input().strip()
ans = 0

for i in range(N):
    if lane1[i] == 'R' and lane2[i] == 'R':
        break
    ans += 1
print(ans)

sys.stdout.close()

File: tasks/2025/round4_pp/twolanes-pp.cpp
// @check-accepted: examples N<=15 lane1=lane2 oneObstacle no-limits
// NOTE: it is recommended to use this even if you don't understand the following code.

#include <fstream>
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int main() {
    // uncomment the two following lines if you want to read/write from files
    // ifstream cin("input.txt");
    // ofstream cout("output.txt");

    int N;
    cin >> N;

    string lane1;
    cin >> lane1;

    string lane2;
    cin >> lane2;

    int ans = N;
    for (int i = 0; i < N; i++) {
        if (lane1[i] == 'R' && lane2[i] == 'R') ans = min(ans, i);
    }

    cout << ans << endl;

    return 0;
}
